public/viewjs/metrics.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Metrics Javascript
+ */
+
+/* Charting */
+var dataPoints = [];
+$("#metrics-table tbody tr").each(function () {
+	var self = $(this);
+	var label = self.find("td:eq(0)").attr('data-chart-label');
+	var value = Number(self.find("td:eq(1)").attr('data-chart-value'));
+	var dataPoint = { label: label, y: parseFloat((Math.round(value * 100) / 100).toFixed(2))};
+	dataPoints.push(dataPoint);
+});
+
+var options = {
+	exportEnabled: true,
+	legend:{
+		horizontalAlign: "center",
+		verticalAlign: "bottom"
+	},
+	data: [{
+		type: "pie",
+		showInLegend: true,
+		toolTipContent: "<b>{label}</b>: ${y} (#percent%)",
+		indexLabel: "{label}",
+		legendText: "{label} (#percent%)",
+		indexLabelPlacement: "outside",
+		valueFormatSTringt: "#,##0.##",
+		dataPoints: dataPoints
+	}]
+};
+
+// needed for recursionCount error
+recursionCount=0;
+$("#metrics-chart").CanvasJSChart(options);
+
+/* DataTables */
+var metricsTable = $('#metrics-table').DataTable({
+	"columnDefs": [
+		{ "type": "num", "targets": 1 }
+	]
+});
+$('#metrics-table tbody').removeClass("d-none");
+metricsTable.columns.adjust().draw();
+
+/* DateRangePicker */
+const urlParams = new URLSearchParams(window.location.search);
+
+var start_date = moment().startOf("month").format('MM/DD/YYYY');
+var end_date = moment().endOf("month").format('MM/DD/YYYY');
+
+if (urlParams.get('start_date')) start_date = moment(urlParams.get('start_date')) ;
+if (urlParams.get('end_date')) end_date = moment(urlParams.get('end_date'));
+
+$('#daterange-filter').daterangepicker({
+	showDropdowns: true,
+	startDate: start_date,
+	endDate: end_date,
+	locale: {
+		"format": 'MM/DD/YYYY',
+		"firstDay": 1
+	},
+	ranges: {
+		'Today': [moment(), moment()],
+		'Yesterday': [moment().subtract(1, 'days'), moment().subtract(1, 'days')],
+		'Last 7 Days': [moment().subtract(6, 'days'), moment()],
+		'Last 14 Days': [moment().subtract(13, 'days'), moment()],
+		'Last 30 Days': [moment().subtract(29, 'days'), moment()],
+		'This Month': [moment().startOf('month'), moment().endOf('month')],
+		'Last Month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')],
+		'This Year': [moment().startOf('year'), moment().endOf('year')],
+		'Last Year': [moment().subtract(1, 'year').startOf('year'), moment().subtract(1, 'year').endOf('year')]
```

Comment:
Range label localization is missing (I guess).

Reply:
What do you mean? I believe it's in the `localizations/strings.pot` these new UI elements that require translation elements.

Reply:
Haven't tested it so far due to the missing view, but in the current state, I don't think `Today`, `Yesterday`, `Last 7 Days` and so on will be translated...

Reply:
Yep, I see what you mean now. How would I "test" the translations? 

I'm adding the keys to the primary `strings.pot` in the root of `localization` folder. 

Then I'm duplicating the keys:values in `de/strings.po` and flushing the cache `rm -rf data/viewcache/*`

config.php -> `DEFAULT_LOCALE=de`

Reply:
If you want to test it locally, then of course this would be a way to do it - I never do that since if the `__t()` function or whatever is broken, most likely also the untranslated text won't display.